### PR TITLE
cmd/openshift-install/create: Use NewListWatchFromClient

### PR DESCRIPTION
**Pull request by wking**
At @staebler's [suggestion][1], allowing us to offload some boilerplate to [an upstream library][2] (although it is a bit sad to drop the typed client :p).

Also fix an "an" -> "a" typo and consolidate the debug logging to avoid claiming "added" for modifications.

CC @sttts

[1]: https://github.com/openshift/installer/pull/1375#discussion_r263145567
[2]: https://godoc.org/k8s.io/client-go/tools/cache#NewListWatchFromClient

### CODE REVIEW

File: cmd/openshift-install/create.go [hunk 1]
```diff
@@ -288,11 +286,9 @@ func destroyBootstrap(ctx context.Context, config *rest.Config, directory string
 		return errors.Wrap(err, "waiting for Kubernetes API")
 	}
 
-	events := client.CoreV1().Events("kube-system")
-
 	eventTimeout := 30 * time.Minute
 	logrus.Infof("Waiting up to %v for the bootstrap-complete event...", eventTimeout)
-	if err := waitForEvent(ctx, events, "bootstrap-complete", eventTimeout); err != nil {
+	if err := waitForEvent(ctx, client.CoreV1().RESTClient(), "event", "kube-system", "bootstrap-complete", eventTimeout); err != nil {
```
Comment (abhinavdahiya) [inline cmd/openshift-install/create.go]: we already know we are waiting for events, what pass `"event"` resource ?

Comment (abhinavdahiya) [reply cmd/openshift-install/create.go]: also are we going to wait for events in namespaces other than `kube-system`. if not overkill for now.

Comment (abhinavdahiya) [inline cmd/openshift-install/create.go]: Also the resource is `events`

Comment (abhinavdahiya) [reply cmd/openshift-install/create.go]:
```console
$ oc get --raw /api/v1/ | jq '.resources[].name' | grep events
"events"
```

@sttts might have a better way to verify.

Comment (wking) [reply cmd/openshift-install/create.go]: > Also the resource is `events`

And I'm missing the typed client more already :p.  Fixed with 30e79f40d -> de62f647a.

Comment (wking) [reply cmd/openshift-install/create.go]: > we already know we are waiting for events, what pass `"event"` resource ?

> also are we going to wait for events in namespaces other than `kube-system`. if not overkill for now.

Fixed both of these in de62f647a -> d3a4d8b31.

**Comment by wking** [comment]
Ooops:

```console
$ curl -s https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_installer/1380/pull-ci-openshift-installer-master-e2e-aws/4406/artifacts/e2e-aws/installer/.openshift_install.log | grep bootstrap-complete | tail -n4
time="2019-03-07T01:01:54Z" level=info msg="Waiting up to 30m0s for the bootstrap-complete event..."
time="2019-03-07T01:12:20Z" level=debug msg=" bootstrap-complete: cluster bootstrapping has completed"
time="2019-03-07T01:28:16Z" level=debug msg=" bootstrap-complete: cluster bootstrapping has completed"
time="2019-03-07T01:31:54Z" level=fatal msg="failed to wait for bootstrap-complete event: timed out waiting for the condition"
```

I'd conflated [`v1.Event`][1] with [`watch.Event`][2] :man_facepalming:  Should be fixed with d3a4d8b31 -> a1a6d2a62.

[1]: https://godoc.org/k8s.io/api/core/v1#Event
[2]: https://godoc.org/k8s.io/apimachinery/pkg/watch#Event

**Pull request merged**